lib/Console/Traits/Parallelization.php

```diff
@@ -0,0 +1,107 @@
+ *
+ * @copyright  Copyright (c) Pimcore GmbH (http://www.pimcore.org)
+ * @license    http://www.pimcore.org/license     GPLv3 and PEL
+ */
+
+namespace Pimcore\Console\Traits;
+
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Command\LockableTrait;
+use Symfony\Component\Console\Input\Input;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Webmozarts\Console\Parallelization\Parallelization as WebmozartParallelization;
+
+trait Parallelization
+{
+    use LockableTrait;
+    
+    use WebmozartParallelization
+    {
+        WebmozartParallelization::configureParallelization as parentConfigureParallelization;
+    }
+
+    protected static function configureParallelization(Command $command): void
+    {
+        $command
+            ->addArgument(
+                'item',
+                InputArgument::OPTIONAL,
+                'The item to process'
+            )
+            ->addOption(
+                'processes',
+                null,
+                //'p', avoid collisions with already existing Pimcore command options
+                InputOption::VALUE_OPTIONAL,
+                'The number of parallel processes to run',
+                1
+            )
+            ->addOption(
+                'child',
+                null,
+                InputOption::VALUE_NONE,
+                'Set on child processes'
+            )
+        ;
+    }
+
+    /**
+     * Default behavior in commands: only allow one command of a type at the same time.
+     */
+    protected function runBeforeFirstCommand(InputInterface $input, OutputInterface $output): void
+    {
+        if (!$this->lock()) {
+            $this->writeError('The command is already running.');
+            exit(1);
+        }
+    }
+
+    /**
+     * Default behavior in commands: clean up garbage after each batch run, if there is only
+     * one master process in place.
+     * @param array $items
+     */
+    protected function runAfterBatch(InputInterface $input, OutputInterface $output, array $items): void
+    {
+        if ((int)$this->input->getOption('processes') <= 1) {
+            if ($output->isVeryVerbose()) {
+                $output->writeln('Collect garbage.');
+            }
+            \Pimcore::collectGarbage();
+        }
+    }
+
+    /**
+     * Default behavior in commands: release lock on termination.
+     */
+    protected function runAfterLastCommand(InputInterface $input, OutputInterface $output): void
+    {
+        $this->release(); //release the lock
+    }
+
+    protected function getItemName(int $count): string
+    {
+        return $count <= 1 ? 'item' : 'items';
+    }
+
+    /**
+     * @return \Symfony\Component\DependencyInjection\ContainerInterface|null
+     */
+    protected function getContainer()
```

Comment:
why would we need that? 
since commands are services anyway, we should be able to inject all container stuff?

Reply:
As far as I know, this method is used by Webmozarts internally, to reset the container for the children. It needs to be implemented, but isn't accessed from "the outside".
There are already plans by Webmozarts to get rid of it in V2.x: 
https://github.com/webmozarts/console-parallelization/issues/1